src/components/Heroes/RankColumns.ts

```diff
@@ -0,0 +1,271 @@
+import heroes from 'dotaconstants/build/heroes.json';
+import constants from 'src/components/constants'
+import { abbreviateNumber, displayHeroId } from '../../utility';
+import { GlobalStrings } from 'src/types/common/GlobalStrings'
+import { Heroes } from 'src/types/Hero/Heroes'
+
+type Props = {
+  tabType: HeroesTabType
+  strings: GlobalStrings
+}
+
+enum HeroesTabType {
+  PRO = 'pro',
+  PUBLIC = 'public'
+}
+
+type ProColumn = {
+  displayName: string,
+  field: string,
+  sortFn: boolean,
+  percentBarsWithValue: (row: Row) => string
+}
+
+type PublicColumn = {
+  displayName: string,
+  field: string,
+  sortFn: boolean,
+  percentBarsWithValue: (row: Row) => string
+  displayIcon: string
+  colColor: string
+}
+
+type HeroColumn = {
+  displayName: string,
+  tooltip: string,
+  field: string
+  displayFn: () => JSX.Element,
```

Comment:
Would this work? 

```suggestion
  displayFn: () => React.ReactNode,
```

Reply:
Yes. Is there a reason you prefer one over the other? I don't actually know the difference

Reply:
NOTE: I actually only have flow background, typescript is a new thing for me.

This [stackoverflow answer](https://stackoverflow.com/a/58123882) seems to have a good explanation about the difference.

tl;dr
1. `JSX.Element` is basically `React.ReactElement<any, any>` within react context.
1. `React.ReactNode` includes this, `string`, `null`, `boolean`, and all other things that can be rendered by react component.
1. It doesn't look like this particular `displayFn` requires child to be exactly an element.

None of these points are strong reason to use `React.ReactElement` or `React.ReactNode` over `JSX.Element`, but in my (flow-biased) opinion, explicit type for react-related item will be a better option

Reply:
Well there you go; TIL. Thanks for that, makes sense